Basic processing documentation for CLI.

Garyfallidis opened the pull request.

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -4,23 +4,80 @@
+    dwi.bval
+    dwi.bvec
+    dwi.nii
+    T1.nii
```
These are not the file-names here:

```
T1.nii
__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bval
__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bvec
__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
```

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -4,23 +4,80 @@
+
+Investigate the bvals::
+
+    $ dipy_info dwi.nii
```
```suggestion
    $ dipy_info __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
```

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -4,23 +4,80 @@
+
+Visualize the diffusion data::
+
+    dipy_horizon __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
```
This currently throws an error: 

```
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_horizon", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_horizon", line 10, in <module>
    run_flow(HorizonFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 90, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/viz.py", line 59, in run
    for input_output in io_it:
  File "/Users/arokem/source/dipy/dipy/workflows/multi_io.py", line 258, in __iter__
    IO = np.concatenate([ins, out], axis=1)
ValueError: all the input arrays must have same number of dimensions
```

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -4,23 +4,80 @@
 
 Create brain mask::
 
     dipy_median_otsu dwi.nii.gz --out_dir out_work/
```
```suggestion
    dipy_median_otsu __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii --out_dir out_work/
```

arokem replied:
Also, the API has changed, so we need to provide a vol_idx: 

```
(dipy) cfin_multib$dipy_median_otsu __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii --out_dir out_work/
INFO:Applying median_otsu segmentation on __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_median_otsu", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_median_otsu", line 9, in <module>
    run_flow(MedianOtsuFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 90, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/segment.py", line 73, in run
    autocrop=autocrop, dilate=dilate)
  File "/Users/arokem/source/dipy/dipy/segment/mask.py", line 196, in median_otsu
    raise ValueError("For 4D images, must provide vol_idx input")
ValueError: For 4D images, must provide vol_idx input
```

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -4,23 +4,80 @@
+
+Perform DTI::
 
     dipy_fit_dti dwi.nii.gz dwi.bval dwi.bvec out_work/brain_mask.nii.gz --out_dir out_work/
```
```suggestion
    dipy_fit_dti __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii.gz __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bval __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bvec out_work/brain_mask.nii.gz --out_dir out_work/
```

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,241 @@
+
+We can use ``dipy_median_otsu`` to build a brain mask for the diffusion data::
+
+    dipy_median_otsu dwi.nii --out_dir out_work/
```
I am getting 
```
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_median_otsu", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_median_otsu", line 7, in <module>
    run_flow(MedianOtsuFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 89, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/segment.py", line 72, in run
    autocrop=autocrop, dilate=dilate)
  File "/Users/arokem/source/dipy/dipy/segment/mask.py", line 195, in median_otsu
    raise ValueError("For 4D images, must provide vol_idx input")
```

arokem replied:
This works when I run: `dipy_median_otsu dwi.nii --out_dir out_work/ --vol_idx 0`

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,241 @@
+We can visualize the result using ``dipy_horizon``. The ``--cluster`` option allows to directly see the clusters of the tractogram::
 
-Create peaks::
+    dipy_horizon out_work/tracts_from_peaks.trk --cluster
```
I am getting: 

```
(dipy) cfin_multib$dipy_horizon out_work/tracts_from_peaks.trk --cluster
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_horizon", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_horizon", line 7, in <module>
    run_flow(HorizonFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 89, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/viz.py", line 104, in run
    for input_output in io_it:
  File "/Users/arokem/source/dipy/dipy/workflows/multi_io.py", line 258, in __iter__
    IO = np.concatenate([ins, out], axis=1)
ValueError: all the input arrays must have same number of dimensions
```

Any ideas?

skoudoro replied:
Sometimes, you have this error when it doesn't find/understand the path... Can you add some double quotes. My last PR improve a bit this point but I think in general we should start to use pathlib module.

arokem replied:
Oh - I get it! It's just a typo. This should be: 

`$ dipy_horizon out_work/tracks_from_peaks.trk --cluster`

arokem commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,241 @@
+Alternatively, we can create deterministic tracks using the maximum value of a spherical harmonics cone::
 
-Create seeding mask (this creates seed_mask in new folder)::
+    dipy_track_det peaks.pam5 fa.nii.gz seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --use_sh
```
I believe that the CLI `dipy_track_det` is now `dipy_track_det`. I am also getting: 

`dipy_track_local: error: unrecognized arguments: --use_sh`

arokem commented:
You should add this file in the index of this section: 

https://github.com/dipy/dipy/blob/master/doc/interfaces/index.rst

And make sense that this and the already-existing example make sense.

skoudoro commented:
Hi @Garyfallidis,

What is the status of this PR? Can you address @arokem comments? Is it close to be done?

Garyfallidis commented:
Hello I have addressed the comments and fixed some other things too.

I think this is okay for now as a basic introduction to our workflows but we should soon used a better (whole brain and without strange filenames) dataset.  For this reason I also opened and issue to not forget. Here it is https://github.com/dipy/dipy/issues/2138

skoudoro commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,248 @@
+    Visualization of a slice from the generated brain mask
+
+
+Perform DTI using ``dipy_fit_dti`. The input of this function is the DWI data, b-values and b-vector files and the brain mask that we calculated in the previous step::
```
typo: 1 back stick to remove on the left of `dipy_fit_dti `

skoudoro commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,248 @@
+We can visualize the result using ``dipy_horizon``. The ``--cluster`` option allows to directly see the clusters of the tractogram::
 
-    dipy_fit_csd dwi.nii.gz dwi.bval dwi.bvec out_work/brain_mask.nii.gz --out_dir out_work/
+    dipy_horizon out_work/tracts_from_peaks.trk --cluster
```
typo: `tracks_from_peaks.trk` instead of `tracts_from_peaks.trk`

skoudoro commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,248 @@
+Alternatively, we can create deterministic tracks using the maximum value of a spherical harmonics cone::
 
-    dipy_mask out_work/fa.nii.gz 0.4 --out_dir out_work/ --out_mask seed_mask.nii.gz
+    dipy_track_local peaks.pam5 fa.nii.gz seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --use_sh
```
`--use_sh` parameter does not exist anymore. You should use `--tracking_method` and choose your method between:
- "eudx" 
- "deterministic" or "det" for a deterministic tracking
- "probabilistic" or "prob" for a Probabilistic tracking
- "closestpeaks" or "cp" for a ClosestPeaks tracking

Furthermore, `peaks.pam5` is not on the current folder, so you need to do `out_work/peaks.pam5`. The correct command line is:
`dipy_track_local out_work/peaks.pam5 out_work/fa.nii.gz out_work/seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --tracking_method eudx`

skoudoro commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,248 @@
+    Frontiers in Neuroinformatics, 1-18, 2014.
 
-    dipy_track_det peaks.pam5 fa.nii.gz seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --use_sh
+.. [Aganj2010] Aganj I, Lenglet C, Sapiro G, Yacoub E, Ugurbil K, Harel N.
```
this citation is never referenced

skoudoro commented on doc/interfaces/basic_flow.rst:
```diff
@@ -1,42 +1,248 @@
+   multiple-shell q-ball imaging within constant solid angle", Magnetic
+   Resonance in Medicine. 2010 Aug;64(2):554-66. doi: 10.1002/mrm.22365
 
+.. [Garyfallidis12] Garyfallidis E., "Towards an accurate brain tractography",
```
this citation is never referenced

Garyfallidis commented:
Thank you for your review. I believe it is corrected now. Let me know otherwise.

skoudoro commented:
Hi @Garyfallidis,

There are still 2 comments not addressed. The command line is failing because of a typo. Can you see above my comment?
Thank you

skoudoro commented:
Thank you @Garyfallidis! merging
